### DISCUSSION TITLE
Lag when doing on change with Multipicker

### DISCUSSION THREAD

**Discussion by vaishnaray** [Budibase/budibase#11640] [Help] [answered]
I am doing an onchange action on a Multipicker form element where if the user selects a value from the picker, it should automatically update the new text field.  This multipicker is in a repeater. 

<img width="267" alt="image" src="https://github.com/Budibase/budibase/assets/9454156/1041e1a3-8ac9-4fcb-a2cb-3b9860cc326f">

This is how I defined my on change action for the picker:
<img width="756" alt="image" src="https://github.com/Budibase/budibase/assets/9454156/4a4397a3-4afe-4342-89d8-52d201427f0e">

However, when I got to click on the values in the picker, it has a delayed response and shows me the value that I have previously selected ONLY after I picked the next value. For example, here I have chosen a value in the multipicker, however the field does not update with that value.

<img width="551" alt="image" src="https://github.com/Budibase/budibase/assets/9454156/a17de500-1c92-4b6b-bf76-5f0a117a1275">

And now when I click on another value in the picker (B), the field updates but only with the previous value ID (A). 

<img width="453" alt="image" src="https://github.com/Budibase/budibase/assets/9454156/2c7dc75b-d6a6-4006-abdf-8ad67dfca225">

How do I get the current values selected into the text field? Any help would be much appreciated!

**Comment by andz-bb** [comment, marked as answer]
hey Vaish,

in this case you need to use the `{{ Field Value }}` binding for the value instead of the field on the form binding. At the point when the on change action is being run the form object will still have the old value while the `{{ Field Value }}` binding has the newly changed value. You'll be able to find it on the bindings sidebar under the name of the field component, in your case Multiselect.

<img width="874" alt="image" src="https://github.com/Budibase/budibase/assets/110921612/bff969cd-aef1-41ea-a720-10fea61bb071">

hope this helps - if you have any questions on this feel free to reach out
Andrew

**Comment by vaishnaray** [comment]
Thank you so much! Will try this.

**Comment by vaishnaray** [comment]
I can't see Multiselect - Field Value in the list when I try to do a
Trigger Automation.

[image: image.png]

[image: image.png]
